module4/Homework4Playground.playground/Contents.swift

```diff
@@ -0,0 +1,211 @@
+import Cocoa
+
+//a) In the assignment for Module 3, part D asked you to write a function that would compute the average of an array of Int. Using that function and the array created in part A, create two overloaded functions of the function average.
+
+print("a) ---------------")
+
+let firstArry = Array(0...20)
+
+func average(_ array: [Int]?) {
+  if let array = array {
+    var sum = 0
+    for el in array {
+      sum += el
+    }
+    let average = array.count > 0 ? Double(sum) / Double(array.count) : 0.0
+    print("The average of the values in the array is \(average).")
+  } else {
+    print("The array is nil. Calculating the average is impossible.")
+  }
+}
+
+
+// take non-optiolnal argument + new arg label
+func average(forNonOptional array: [Int]) {
+  print("calling average() overload nonOptional parameter")
+  average(array)
+}
+
+// average with default parameter
+func average(array: [Int]? = firstArry) {
+  print("calling average() overload with default parameter")
+  average(array)
+}
+
```

Comment:
these just call the original. What is also common is to replace the original to extend the function. For example
````
func average(_ array: [Double]) {
let total = array.reduce(0.0, +)
print("The average of the values in the array is \(total/Double(array.count))")
}
````